keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/artifact/ImageHandler.kt

```diff
@@ -0,0 +1,160 @@
+        // we clearly care about this artifact, let's register it.
+        artifactRepository.register(this)
+        publisher.publishEvent(ArtifactRegisteredEvent(this))
+      }
+    }
+
+    // even though the artifact isn't registered we should grab the latest version to use
+    val versions = igorService
+      .getVersions(name, statuses.map { it.toString() })
+    log.debug("Finding latest version of $name: versions igor knows about = $versions")
+    return versions
+      .firstOrNull()
+      ?.let {
+        val version = "$name-$it"
+        log.debug("Finding latest version of $name, choosing = $version")
+        version
+      } ?: throw NoKnownArtifactVersions(this)
+  }
+
+  private suspend fun launchBake(
+    artifact: DebianArtifact,
+    desiredVersion: String
+  ): List<Task> {
+    val appVersion = AppVersion.parseName(desiredVersion)
+    val packageName = appVersion.packageName
+    val version = desiredVersion.substringAfter("$packageName-")
+    val artifactRef = "/${packageName}_${version}_all.deb"
+    val artifactPayload = mapOf(
+      "type" to "DEB",
+      "customKind" to false,
+      "name" to artifact.name,
+      "version" to version,
+      "location" to "rocket",
+      "reference" to artifactRef,
+      "metadata" to emptyMap<String, Any>(),
+      "provenance" to "n/a"
+    )
+
+    log.info("baking new image for {}", artifact.name)
+    val description = "Bake $desiredVersion"
+
+    val (serviceAccount, application) = artifact.taskAuthenticationDetails
+
+    try {
+      val taskRef = taskLauncher.submitJob(
+        user = serviceAccount,
+        application = application,
+        notifications = emptySet(),
+        subject = description,
+        description = description,
+        correlationId = artifact.correlationId,
+        stages = listOf(
+          Job(
+            "bake",
+            mapOf(
+              "amiSuffix" to "",
+              "baseOs" to artifact.vmOptions.baseOs,
+              "baseLabel" to artifact.vmOptions.baseLabel.name.toLowerCase(),
+              "cloudProviderType" to "aws",
+              "package" to artifactRef.substringAfterLast("/"),
+              "regions" to artifact.vmOptions.regions,
+              "storeType" to artifact.vmOptions.storeType.name.toLowerCase(),
+              "user" to "keel",
+              "vmType" to "hvm"
+            )
+          )
+        ),
+        artifacts = listOf(artifactPayload)
+      )
+      return listOf(Task(id = taskRef.id, name = description))
+    } catch (e: Exception) {
+      log.error("Error launching bake for: ${description.toLowerCase()}")
+      return emptyList()
+    }
+  }
+
+  private val DebianArtifact.taskAuthenticationDetails: BakeCredentials
+    get() = deliveryConfigName?.let {
+      deliveryConfigRepository.get(it).run {
+        BakeCredentials(serviceAccount, application)
+      }
+    } ?: defaultCredentials
+
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+}
```

Comment:
This is the meat of it -- totally new version of `ImageHandler` that looks at the artifact rather than the `Resource<ImageSpec>`